server/issue.go

```diff
@@ -733,126 +733,90 @@ func (s *Server) getPipelineCreateForDatabaseSchemaAndDataUpdate(ctx context.Con
 	}
 
-	// Tenant mode project pipeline has its own generation.
-	if project.TenantMode == api.TenantModeTenant {
-		if !s.feature(api.FeatureMultiTenancy) {
-			return nil, echo.NewHTTPError(http.StatusForbidden, api.FeatureMultiTenancy.AccessErrorMessage())
+	// Validate issue detail list.
+	if len(c.DetailList) == 0 {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, "Tenant mode project should have at least one update schema detail")
+	}
+	databaseNameCount, databaseIDCount := 0, 0
+	for _, detail := range c.DetailList {
+		if detail.MigrationType != db.Migrate && detail.MigrationType != db.Data {
+			return nil, echo.NewHTTPError(http.StatusBadRequest, "Only Migrate and Data type migration can be performed on tenant mode project")
 		}
-		if len(c.DetailList) == 0 {
-			return nil, echo.NewHTTPError(http.StatusBadRequest, "Tenant mode project should have at least one update schema detail")
+		if detail.Statement == "" {
+			return nil, echo.NewHTTPError(http.StatusBadRequest, "Failed to create issue, sql statement missing")
 		}
-		databaseNameCount, databaseIDCount := 0, 0
-		for _, detail := range c.DetailList {
-			if detail.MigrationType != db.Migrate && detail.MigrationType != db.Data {
-				return nil, echo.NewHTTPError(http.StatusBadRequest, "Only Migrate and Data type migration can be performed on tenant mode project")
-			}
-			if detail.Statement == "" {
-				return nil, echo.NewHTTPError(http.StatusBadRequest, "Failed to create issue, sql statement missing")
-			}
-			if detail.DatabaseID > 0 {
-				databaseIDCount++
-			}
-			if detail.DatabaseName != "" {
-				databaseNameCount++
-			}
+		if detail.DatabaseID > 0 {
+			databaseIDCount++
 		}
-		if databaseNameCount > 0 && databaseIDCount > 0 {
-			return nil, echo.NewHTTPError(http.StatusBadRequest, "Migration detail should set either database name or database ID.")
+		if detail.DatabaseName != "" {
+			databaseNameCount++
 		}
-		if databaseNameCount > 1 {
-			return nil, echo.NewHTTPError(http.StatusBadRequest, "There should be at most one migration detail with database name.")
+	}
+	if databaseNameCount > 0 && databaseIDCount > 0 {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, "Migration detail should set either database name or database ID.")
+	}
+	if databaseNameCount > 1 {
+		return nil, echo.NewHTTPError(http.StatusBadRequest, "There should be at most one migration detail with database name.")
+	}
+	if project.TenantMode == api.TenantModeTenant && !s.feature(api.FeatureMultiTenancy) {
+		return nil, echo.NewHTTPError(http.StatusForbidden, api.FeatureMultiTenancy.AccessErrorMessage())
+	}
+	maximumTaskLimit := s.getPlanLimitValue(api.PlanLimitMaximumTask)
+	if databaseIDCount > 0 && int64(databaseIDCount) > maximumTaskLimit {
+		return nil, echo.NewHTTPError(http.StatusForbidden, fmt.Sprintf("Effective plan %s can update up to %d databases, got %d.", s.getEffectivePlan(), maximumTaskLimit, len(c.DetailList)))
```

Comment:
error log uses len(c.DetailList) which is different from int64(databaseIDCount)

Reply:
Done.